zkevm-circuits/src/evm_circuit/op_execution/push.rs

```diff
@@ -0,0 +1,306 @@
+                selector: case_selector.exp(),
+                linear_combinations: [
+                    op_execution_state_transition_constraints,
+                    push_constraints,
+                ]
+                .concat(),
+                lookups: [byte_range_lookup, bus_mapping_lookups].concat(),
+            }
+        };
+
+        let stack_overflow = {
+            let (zero, minus_one) = (
+                Expression::Constant(F::from_u64(1024)),
+                Expression::Constant(F::from_u64(1023)),
+            );
+            let stack_pointer = op_execution_state_curr.stack_pointer.exp();
+            Constraint {
+                name: "PushGadget stack overflow",
+                selector: self.stack_overflow.exp(),
+                linear_combinations: vec![
+                    (stack_pointer.clone() - zero)
+                        * (stack_pointer - minus_one),
+                ],
+                lookups: vec![],
+            }
+        };
+
+        let out_of_gas = {
+            let (one, two, three) = (
+                Expression::Constant(F::from_u64(1)),
+                Expression::Constant(F::from_u64(2)),
+                Expression::Constant(F::from_u64(3)),
+            );
+            let (case_selector, gas_available) = &self.out_of_gas;
+            let gas_overdemand = op_execution_state_curr.gas_counter.exp()
+                + three.clone()
+                - gas_available.exp();
+            Constraint {
+                name: "PushGadget out of gas",
+                selector: case_selector.exp(),
+                linear_combinations: vec![
+                    (gas_overdemand.clone() - one)
+                        * (gas_overdemand.clone() - two)
+                        * (gas_overdemand - three),
+                ],
+                lookups: vec![],
+            }
+        };
+
+        vec![common, success, stack_overflow, out_of_gas]
+    }
+
+    fn assign(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        op_execution_state: &mut OpExecutionStateInstance,
+        execution_step: &ExecutionStep,
+    ) -> Result<(), Error> {
+        match execution_step.case {
+            Case::Success => self.assign_success(
+                region,
+                offset,
+                op_execution_state,
+                execution_step,
+            ),
+            Case::StackOverflow => {
+                unimplemented!()
+            }
+            Case::OutOfGas => {
+                unimplemented!()
+            }
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl<F: FieldExt> PushGadget<F> {
+    fn assign_success(
+        &self,
+        region: &mut Region<'_, F>,
+        offset: usize,
+        op_execution_state: &mut OpExecutionStateInstance,
+        execution_step: &ExecutionStep,
+    ) -> Result<(), Error> {
+        op_execution_state.global_counter += 1;
+        op_execution_state.program_counter +=
```

Comment:
push1 operation increases program_counter + 2 which is different than other op code

Reply:
Fixed. Thanks!